cpp/src/arrow/util/compression_benchmark.cc

```diff
@@ -133,6 +132,38 @@ static void ReferenceStreamingCompression(
   StreamingCompression(COMPRESSION, data, state);
 }
 
+int64_t NonStreamingCompress(Codec* codec, const std::vector<uint8_t>& data,
```

Comment:
How about removing `NonStreaming` because `ReferenceCompression` doesn't have `NonStreaming` prefix?

```suggestion
int64_t Compress(Codec* codec, const std::vector<uint8_t>& data,
```